Classes/DDFileLogger.m

```diff
@@ -1046,7 +1046,7 @@ - (BOOL)lt_shouldUseLogFile:(nonnull DDLogFileInfo *)logFileInfo isResuming:(BOO
     // If we're resuming, we need to check if the log file is allowed for reuse or needs to be archived.
     if (isResuming && (_doNotReuseLogFiles || [self lt_shouldLogFileBeArchived:logFileInfo])) {
         logFileInfo.isArchived = YES;
-        NSString *archivedLogFilePath = [logFileInfo.fileName copy];
+        NSString *archivedLogFilePath = [logFileInfo.filePath copy];
```

Comment:
Can you move the line inside the `[_logFileManager respondsToSelector:@selector(didArchiveLogFile:)]`test. There is no need to copy the file path if the manager doesn't respond to `-didArchiveLogFile:`.

Reply:
BTW you can do the same by moving the file path copy when invoking `-didRollAndArchiveLogFile:`.